docs/source/05_data/02_kedro_io.md

```diff
@@ -459,13 +459,36 @@ def create_partitions() -> Dict[str, Any]:
     return {
         # create a file "s3://my-bucket-name/part/foo.csv"
         "part/foo": pd.DataFrame({"data": [1, 2]}),
-        # create a file "s3://my-bucket-name/part/bar.csv.csv"
+        # create a file "s3://my-bucket-name/part/bar.csv"
         "part/bar.csv": pd.DataFrame({"data": [3, 4]}),
     }
 ```
 
 > *Note:* Writing to an existing partition may result in its data being overwritten, if this case is not specifically handled by the underlying dataset implementation. You should implement your own checks to ensure that no existing data is lost when writing to a `PartitionedDataSet`. The simplest safety mechanism could be to use *partition IDs* that have a high chance of uniqueness - for example, the current timestamp.
 
+`PartitionedDataSet` also supports lazy saving, where the partition's data is not materialized until it's time to write.
+To use this, simply return `Callable` types in the dictionary:
+
+```python
+from typing import Any, Dict, Callable
+import pandas as pd
+
+def create_partitions() -> Dict[str, Callable[[], Any]]:
+    """Create new partitions and save using PartitionedDataSet.
+
+    Returns:
+        Dictionary of the partitions to create to a function that creates them.
+    """
+    return {
+        # create a file "s3://my-bucket-name/part/foo.csv"
+        "part/foo": lambda: pd.DataFrame({"data": [1, 2]}),
+        # create a file "s3://my-bucket-name/part/bar.csv"
+        "part/bar.csv": lambda: pd.DataFrame({"data": [3, 4]}),
+    }
+```
+
+> *Note:* Lazy saving causes the dataset to be written _after_ the `after_node_run` [hook](../07_extend_kedro/02_hooks), which could cause unintended behavior. 
```

Comment:
Is there anything you could do to avoid this potential unintended behaviour? If not, perhaps it's better to just rephrase this to: *Note:* When using lazy saving the dataset will be written _after_ the `after_node_run` [hook](../07_extend_kedro/02_hooks).

Otherwise it kind of sounds like there is maybe something you could do differently.

Reply:
I don't believe there is a way to avoid this (I took this note from the original issues discussion). I'll update the documentation to make this clearer. Thanks so much!